lib/routes/lnurl_fetch_invoice_page.dart

```diff
@@ -0,0 +1,376 @@
+            content // base64 string, up to 136536 characters (100Kb of image data in base-64 encoding)
+        ],
+        [
+            "image/jpeg;base64", // optional 512x512px JPG thumbnail which will represent this lnurl in a list or grid
+            content // base64 string, up to 136536 characters (100Kb of image data in base-64 encoding)
+        ],
+        ... // more objects for future types
+    ]
+    ```
+
+    and be sent as a string:
+
+    ```
+    "[[\"text/plain\", \"lorem ipsum blah blah\"]]"
+    ```
+    */
+    var result = '';
+    log.info('_getMetadataText: metadata: $metadata');
+
+    for (var datum in metadata) {
+      if (datum.entry[0] == 'text/plain') {
+        result = datum.entry[1];
+        break;
+      }
+    }
+
+    return result;
+  }
+
+  void _getInvoice(InvoiceBloc invoiceBloc, AccountBloc accountBloc,
+      LNUrlBloc lnurlBloc, AccountModel account) async {
+    log.info('_getInvoice.');
+    /*
+         5. LN WALLET makes a GET request using callback with the following query parameters:
+         amount (input) - user specified sum in MilliSatoshi
+         nonce - an optional parameter used to prevent server response caching
+         fromnodes - an optional parameter with value set to comma separated nodeIds if payer wishes a service to provide payment routes starting from specified LN nodeIds
+         comment (input) - an optional parameter to pass the LN WALLET user's comment to LN SERVICE
+
+    */
+
+    var navigator = Navigator.of(context);
+    _currentRoute = ModalRoute.of(navigator.context);
+    _loaderRoute = createLoaderRoute(context);
+    navigator.push(_loaderRoute);
+
+    _payFetchResponse.amount =
+        account.currency.parse(_amountController.text) * 1000;
+    _payFetchResponse.comment = _commentController.text;
+
+    LNUrlPayInfo payinfo;
+    var action = FetchInvoice(_payFetchResponse);
+    action.future.catchError((e) {
+      promptError(
+          context,
+          "LNURL-Pay Error",
+          Text(
+              "An error occurred while attempting to retreive an invoice from ${_payFetchResponse.host}!\nReason: ${e.toString()}",
+              style: Theme.of(context).dialogTheme.contentTextStyle),
+          okFunc: removeLoader);
+    }).then((payinfo) {
+      invoiceBloc.decodeInvoiceSink.add(payinfo.invoice);
+      log.info(
+          '_getInvoice: Found LNUrlPayInfo. Beginning payment request flow.');
+      removeLoader();
+
+      /*
+           FIXME Verify that this works when:
+           "Before popping here, I suggest to check if the user cancelled by closing the dialog (back on android).  One way to do it is to check if the current widget is disposed and continue with the invoice only if it is not." (@roeierez) 
+      */
+
+      if (mounted && _currentRoute.isCurrent) {
+        var navigator = Navigator.of(context);
+        navigator.pop();
+      }
```

Comment:
@roeierez 

re: https://github.com/breez/breezmobile/pull/493/#discussion_r651738611

Is this correct?

Reply:
I think in case the _currentRoute is not _current_ anymore it means that the user has pressed back and probably meant to cancel the flow.
In that case for me it seems more correct to just return and do nothing.
Maybe similar to this:

```
if (!_currentRoute.isCurrent) {
   return;
}
Navigator.of(context).pop();
```

Does it make sense to you as well?